Contributions/Linux_Memory_Management_Essentials.md

```diff
@@ -0,0 +1,249 @@
+
+
+## **Terms and Abbreviations**
+Plese refer to the Linux Kernel documentation.
+   
+
+## **References**
+
+1. [Linux Kernel website](https://www.kernel.org) - <https://www.kernel.org>
+2. ***Interference Scenarios for an ARM64 Linux System***
+3. [CC BY-SA 4.0 Deed | Attribution-ShareAlike 4.0 International | Creative Commons](https://creativecommons.org/licenses/by-sa/4.0/) - <https://creativecommons.org/licenses/by-sa/4.0/> License
+
+
+## **Disclaimer**
+* This document is not intended to be a replacement for understanding the memory management of the Linux Kernel,
+  nor it attempts to be an exhaustive analysis of safety implications.
+* Because of the very volatile nature of the code within the Linux Kernel, each of the statements made
+  below should not be taken at face value, but rather verified, for any Linux kernel version following
+  the one used while writing the document (6.9).
+* When referring to specific HW features, the document refers to the ARM64 architecture.
+
+## **Purpose of the document**
+This document aims to provide an holistic view of what happens in Linux memory management, so that
+one is at least aware of certain features and can use this document as a jumping pad toward more detailed documentation.
+Or even to the code base.
+
+## **Structure of the document**
+The document is divided in two parts, based on the destinatary of the memory allocations discussed: kernel-space or user-space.
+Individual points are numbered for ease of reference, but the numbering is not meant to represent any sequence.
+
+## **Memory management in Linux**
+
+### **Kernel-space memory allocations**
+
+#### **Directly Verifiable Assertions**
+The following section presents a set of statements that can be objectively verified e.g. by inspecting the sources.
+
+1. unlike processes memory, kernel memory pages are not swapped, nor dropped silently by the kernel itself,
+   although an hypervisor will do to a VM what the kernel does to a process (but this is beyond the control of the kernel)
+2. the kernel context (usually EL1 on ARM64) uses one single memory map (page tables) across all the cores
+   executing in kernel mode
+3. on 64 bit systems (e.g. ARM64 and x86_64), usually almost all physical memory is mapped in a
+   (semi)contiguous (there can be holes) range. Memory within this range is both virtually and physically contiguous.
+4. physically contiguous memory is treated as a scarce resource, and typically is not provided to userspace,
+   unless it explicitly asks for it (e.g. for implementing some DMA buffer)
+5. the kernel can access userspace memory in three ways:
+   1. through the userspace mapping 
+      1. this type of access is limited to few functions, like copy_to_user()/copy_from_user() and put_user()/get_user()
+      2. outside the execution of the functions mentioned, this type of access is not possible, because the userspace mappings are made available only while executing such functions
```

Comment:
I would say "outside the execution of the functions mentioned, this type of access would be an unexpected interference, because...", because to make it "not possible" you need HW protections as the ones you mentioned below

Reply:
At least on ARM64 the access through user space mappings is disabled, unless TTBR0 is accidentally enabled (and that would count as 1 interference); to get interference toward the content of the userspace process memory, a rogue operation toward user-space address range shuold happen (and that would count as 2nd interference).

So, my reasoning is that if we consider only 1 interference event, then it is not possible.